connect_deprish_to_ferc1.py

```diff
@@ -0,0 +1,214 @@
+    missing_plant_ids = (steam_cleaned_ferc1[
+        steam_cleaned_ferc1.plant_id_pudl.isin(no_ferc_plant_ids)]
+        .plant_id_pudl.unique())
+    logger.info(f"Connected plants:    {len(connected_plant_ids)}")
+    logger.info(f"Not connected:       {len(no_ferc_plant_ids)}")
+    logger.info(f"Missing connections: {len(missing_plant_ids)}")
+    # Investigation of 3 missing connections [1204, 1147, 1223] determined that
+    # these plants are missing from ferc because there was no reporting
+    # for the specific years in question
+    if len(missing_plant_ids) > 3:
+        warnings.warn(
+            f'There are {len(missing_plant_ids)} missing plant records.')
+
+
+def connect(inputs):
+    """
+    Connect depreciation records with ferc1 steam records.
+
+    TODO: This is a big messy WIP function right now.
+
+    Args:
+        inputs (object)
+
+    Returns:
+        pandas.DataFrame
+    """
+    inputs.prep_inputs()
+
+    connects_all_deprish_ferc1 = pd.merge(
+        inputs.connects_deprish_eia,
+        inputs.steam_cleaned_ferc1,
+        on=['plant_id_pudl',
+            'utility_id_pudl',
+            'report_date'],
+        suffixes=('_deprish', '_ferc1'),
+        how='left', indicator=True
+    )
+    # reorder cols so they are easier to see, maybe remove later
+    first_cols = ['plant_part_deprish', 'plant_part_ferc1']
+    connects_all_deprish_ferc1 = connects_all_deprish_ferc1[
+        first_cols + [x for x in connects_all_deprish_ferc1.columns
+                      if x not in first_cols]]
+
+    check_high_level_connections(
+        connects_all_deprish_ferc1, inputs.steam_cleaned_ferc1)
+
+    replace_dict = {
+        x:
+        f"{inputs.plant_parts_ordered.index(x)}_{x}"
+        for x in inputs.plant_parts_ordered
+    }
+
+    connects_all_deprish_ferc1 = (
+        connects_all_deprish_ferc1.assign(
+            part_no_deprish=lambda x:
+                x.plant_part_deprish.replace(replace_dict),
+            part_no_ferc1=lambda x:
+                x.plant_part_ferc1.replace(replace_dict),
+            level_deprish=lambda x:
+                np.where(
+                    x.part_no_deprish == x.part_no_ferc1,
+                    'samezies',
+                    np.where(x.part_no_deprish < x.part_no_ferc1,
+                             'beeeg',
+                             np.where(x.part_no_ferc1.isnull(),
+                                      'who knows??', 'smol')
+                             )))
+        .drop(columns=['part_no_deprish', 'part_no_ferc1'])
+    )
+
+    smol = connects_all_deprish_ferc1[
+        connects_all_deprish_ferc1.level_deprish == 'smol']
+    beeeg = connects_all_deprish_ferc1[
+        connects_all_deprish_ferc1.level_deprish == 'beeeg']
+    same = connects_all_deprish_ferc1[
+        connects_all_deprish_ferc1.level_deprish == 'samezies']
+    nah = connects_all_deprish_ferc1[
+        connects_all_deprish_ferc1.level_deprish == 'who knows??']
```

Comment:
`df.column_name.value_counts()` is a short way to get these counts.

Reply:
Also, the names are cute, but having some of them be consistent (`beeeg`) and other inconsistent (`same` vs `samezies`) will be extremely frustrating later on.